Spice/src/SpiceClientBuilder.cs

```diff
@@ -80,6 +80,17 @@ public SpiceClientBuilder WithMaxRetries(int maxRetries)
         return this;
     }
 
+    /// <summary>
+    /// Sets the client's user agent.
+    /// </summary>
+    /// <param name="userAgent">User agent string</param>
+    /// <returns>The current instance of <see cref="SpiceClientBuilder"/> for method chaining.</returns>
+    public SpiceClientBuilder WithUserAgent(string userAgent)
+    {
+        _spiceClient.UserAgent = userAgent;
```

Comment:
Might be good to check if it is not an empty string

Reply:
We're not doing any validation of the user-agent string provided by the users. It's entirely up to them